internal/contentmanager/contentmanager.go

```diff
@@ -54,23 +61,46 @@ func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller
 
 	// TODO: add a http.RoundTripper to the config to ensure it is always using an up
 	// to date authentication token for the ServiceAccount token provided in the ClusterExtension.
-	// Maybe this should be handled by the RestConfigMapper?
+	// Maybe this should be handled by the RestConfigMapper
+
+	// Assumptions: all objects received by the function will have the Object metadata specfically,
+	// ApiVersion and Kind set. Failure to which the code will panic when adding the types to the scheme
+	scheme := runtime.NewScheme()
+	for _, obj := range objs {
+		gvk := obj.GetObjectKind().GetVersionKind()
+		listKind := obj.GetObjectKind().GetVersionKind().Kind + "List"
+
+		if gvk.Kind == "" || gvk.Version == "" {
+			return errors.New("object Kind or Version is not defined")
+		}
+
+		if !scheme.Recognizes(gvk) {
+			u := &unstructured.Unstructured{}
+			u.SetGroupVersionKind(gvk)
+			ul := &unstructured.UnstructuredList{}
+
+			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
+
+			scheme.AddKnownTypeWithName(gvk, u)
+			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
+			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
```

Comment:
Looks like the `metav1` import is missing. It should be `metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
`